server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -272,6 +272,17 @@ public static IndexMergePolicy fromString(String text) {
         Property.IndexScope
     );
 
+    /**
+     * Index setting describing the maximum number of nested scopes in queries.
+     * The default maximum of 20. 1 means once nesting.
+     */
+    public static final Setting<Integer> MAX_NESTED_QUERY_DEPTH_SETTING = Setting.intSetting(
+        "index.query.max_nested_depth",
+        20,
```

Comment:
We would need to make a change here (I believe):
```suggestion
        Integer.MAX_VALUE,
```

Provided the context here https://github.com/opensearch-project/documentation-website/pull/6347/files#r1485277588 but TLDR; - it is a breaking change and we could not introduce breaking changes in minor / patch releases.